cargo-shuttle/src/main.rs

```diff
@@ -106,6 +109,38 @@ impl Shuttle {
         .context("failed to delete deployment")
     }
 
+    async fn init(&self, args: InitArgs) -> Result<()> {
+        // Call `cargo init --lib` to generate files
+        let path = args.path.to_str().unwrap();
+        process::Command::new("cargo")
+            .args(["init", "--lib", path])
+            .spawn()?
+            .wait()?;
+
+        let cargo_path = args.path.join("Cargo.toml");
+        let mut cargo_doc = read_to_string(cargo_path.clone())?.parse::<Document>()?;
+
+        // Remove empty dependencies table to re-insert after the lib table is inserted
+        cargo_doc.remove("dependencies");
+
+        // Insert `crate-type = ["cdylib"]` array into `[lib]` table
+        let crate_type_array = Array::from_iter(["cdylib"].into_iter());
+        let mut lib_table = Table::new();
+        lib_table["crate-type"] = Item::Value(Value::Array(crate_type_array));
+        cargo_doc["lib"] = Item::Table(lib_table);
+
+        // Insert shuttle-service to `[dependencies]` table
+        let mut dep_table = Table::new();
+        dep_table["shuttle-service"]["version"] = value(env!("CARGO_PKG_VERSION"));
```

Comment:
I might be wrong, but won't this get the package version of `cargo-shuttle` while we want the package version of `shuttle-service`?

Maybe we can add a constant to `shuttle-service` which gets its value from `CARGO_PKG_VERSION` and then use that constant here.